state/factory/workingset.go

```diff
@@ -147,6 +150,9 @@ func (ws *workingSet) runAction(
 	if protocol.MustGetBlockCtx(ctx).GasLimit < protocol.MustGetActionCtx(ctx).IntrinsicGas {
 		return nil, action.ErrGasLimit
 	}
+	if !protocol.MustGetFeatureCtx(ctx).EnableWeb3Rewarding && isWeb3RewardingAction(elp) {
+		return nil, errUnsupportWeb3Rewarding
+	}
```

Comment:
also add this check into API code, search `validateChainID` you will see how it is used